saleor/plugins/vatlayer/plugin.py

```diff
@@ -178,17 +178,63 @@ def calculate_checkout_line_total(
         discounts: List["DiscountInfo"],
         previous_value: TaxedMoney,
     ) -> TaxedMoney:
+        unit_price = self.__calculate_checkout_line_unit_price(
+            address,
+            discounts,
+            variant,
+            collections,
+            channel,
+            channel_listing,
+            previous_value,
+        )
+        return (
+            unit_price * checkout_line.quantity
+            if unit_price is not None
+            else previous_value
+        )
+
+    def calculate_checkout_line_unit_price(
+        self,
+        checkout: "Checkout",
+        checkout_line: "CheckoutLine",
+        address: Optional["Address"],
+        discounts: Iterable["DiscountInfo"],
+        variant: "ProductVariant",
+        collections: List["Collection"],
+        channel: "Channel",
+        channel_listing: "ProductVariantChannelListing",
+        previous_value: TaxedMoney,
+    ) -> TaxedMoney:
+        unit_price = self.__calculate_checkout_line_unit_price(
+            address,
+            discounts,
+            variant,
+            collections,
+            channel,
+            channel_listing,
+            previous_value,
+        )
+        return unit_price if unit_price is not None else previous_value
+
+    def __calculate_checkout_line_unit_price(
+        self,
+        address: Optional["Address"],
+        discounts: Iterable["DiscountInfo"],
+        variant: "ProductVariant",
+        collections: List["Collection"],
+        channel: "Channel",
+        channel_listing: "ProductVariantChannelListing",
+        previous_value: TaxedMoney,
+    ):
         if self._skip_plugin(previous_value):
-            return previous_value
+            return
 
+        product = variant.product
```

Comment:
We should pass the product to the plugin instead of getting it from the database.